src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,217 @@
+        BUGSNAG_HEADERS.put("Content-Type", "application/json");
+    }
+
+    /**
+     * Build the event request filter that will be passed to Bugsnag with every event data request.
+     */
+    private static void buildEventRequestFilter() {
+        final ObjectMapper mapper = new ObjectMapper();
+        ObjectNode reportingWindowCondition = mapper.createObjectNode();
+        reportingWindowCondition.put("type", "eq");
+        reportingWindowCondition.put("value", BUGSNAG_REPORTING_WINDOW_IN_DAYS + "d");
+
+        ArrayNode sinceFilters = mapper.createArrayNode();
+        sinceFilters.add(reportingWindowCondition);
+
+        ObjectNode filters = mapper.createObjectNode();
+        filters.set("event.since", sinceFilters);
+
+        eventRequestFilter = mapper.createObjectNode();
+        eventRequestFilter.set("filters", filters);
+    }
+
+    /**
+     * Make the initial event data request to Bugsnag. This triggers an asynchronous job to prepare the data for one
+     * single download. The returned event data request can be used to check the status of the request.
+     */
+    public static BugsnagEventRequest makeEventDataRequest() {
+
+        if (ORGANIZATION == null) {
+            LOG.error("Required organization is not available. Unable to make event data request.");
+            return null;
+        }
+
+        URI uri = HttpUtils.buildUri(
+            BUGSNAG_API_URL,
+            "/organizations/" + ORGANIZATION.id + "/event_data_requests",
+            null);
+
+        String response = HttpUtils.httpRequest(
+            uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.POST,
+            BUGSNAG_HEADERS,
+            eventRequestFilter.toString());
+
+        return JsonUtils.getPOJOFromJSON(response, BugsnagEventRequest.class);
+    }
+
+    /**
+     * Get a previously created event data request. A status of 'complete' signals that the requested data is ready to
+     * be downloaded from the populated url parameter.
+     */
+    public static BugsnagEventRequest getEventDataRequest(BugsnagEventRequest originalEventDataRequest) {
+
+        if (ORGANIZATION == null) {
+            LOG.error("Required organization is not available. Unable to get event data request.");
+            return null;
+        }
+
+        String endpoint = "/organizations/" + ORGANIZATION.id + "/event_data_requests/" +
+            originalEventDataRequest.eventDataRequestId;
+
+        URI uri = HttpUtils.buildUri(BUGSNAG_API_URL, endpoint, null);
+
+        String response = HttpUtils.httpRequest(
+            uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.GET,
+            BUGSNAG_HEADERS,
+            null);
+
+        return JsonUtils.getPOJOFromJSON(response, BugsnagEventRequest.class);
+    }
+
+    /**
+     * Get a single organization matching the organization held in the property file.
+     */
+    private static void setOrganization() {
```

Comment:
Do we only call this method so that we can access the organization id property? If so, could we just get rid of all of this code and just set the organization id in the config file?

Reply:
Yes, however, it is not possible to get the organization id from the Bugsnag UI and it is needed for most API calls. The only way to get this is to make a call to the '/user/organizations/' API and extract the required organization based on the name given in the config file.